CorsixTH/Lua/hospital.lua

```diff
@@ -1399,7 +1399,7 @@ function Hospital:receiveMoneyForTreatment(patient)
     else
       reason = _S.transactions.cure_colon .. " " .. casebook.disease.name
     end
-    local amount = self:getTreatmentPrice(disease_id)
+    local amount = patient.pay_amount or self:getTreatmentPrice(disease_id)
```

Comment:
Note that in lua unlike javascript, 0 is truthy, so this doesn't guard against agreeToPay not being called and paying out 0.

Not blocking; just a question of intention.

Reply:
The guard to implement wouldn't be as plainly simple as just a ternary operator as far as I can tell. You'd probably need to conduct two separate checks to avoid a lengthy convoluted and/or statement.

Reply:
```suggestion
    local amount = patient.pay_amount or 0
    amount = amount > 0 and amount or self:getTreatmentPrice(disease_id)
```
Would that work as a guard?

Reply:
Yes, or if patient.pay_amount is always numeric (which isn't yet true, it's set on new patient but missing afterLoad) then just your second line.